### ISSUE TITLE
OSI Stack Simulation Level

### ISSUE THREAD

**Issue by tayloreisman16** [aligungr/UERANSIM#368]
Hey guys,

I want to know what level of the OSI stack is simulated in UERANSIM. I realize that MAC addresses are not yet a part of UERANSIM; therefore, I assume that the MAC layer is not yet defined. If this is the case, how do you deal with HARQ? Does the UE communicate with the RAN via packets?

Thanks,

Taylor

**Comment by aligungr** [comment]
Hi @tayloreisman16 

Following layers are not implemented and they are simulated in UERANSIM:
- PHY
- MAC
- RLC
- PDCP
- SDAP

---

> how do you deal with HARQ? Does the UE communicate with the RAN via packets?
Yes radio layer communication is simulated over UDP packets.

Best.

**Comment by tayloreisman16** [comment]
Hey @aligungr ,

Thanks for the reply.

I noticed that you say the RLC is not yet implemented, but the feature set says it's done? Could you clarify? 

Thank you!

**Comment by aligungr** [comment]
@tayloreisman16 

Yes RLC is done, but it's not currently used in the project. Because RLC protocol is somewhat meaningless without other radio layers.

Best.
